engine/websocketroutine_manager.go

```diff
@@ -106,54 +106,47 @@ func (m *WebsocketRoutineManager) websocketRoutine() {
 	if err != nil {
 		log.Errorf(log.WebsocketMgr, "websocket routine manager cannot get exchanges: %v", err)
 	}
-	wg := sync.WaitGroup{}
-	wg.Add(len(exchanges))
-	for i := range exchanges {
-		go func(i int) {
-			defer wg.Done()
-			if exchanges[i].SupportsWebsocket() {
-				if m.verbose {
-					log.Debugf(log.WebsocketMgr,
-						"Exchange %s websocket support: Yes Enabled: %v",
-						exchanges[i].GetName(),
-						common.IsEnabled(exchanges[i].IsWebsocketEnabled()),
-					)
-				}
+	var wg sync.WaitGroup
+	for _, exch := range exchanges {
+		if !exch.SupportsWebsocket() {
+			if m.verbose {
+				log.Debugf(log.WebsocketMgr, "Exchange %s websocket support: No",
+					exch.GetName())
+			}
+			continue
+		}
 
-				ws, err := exchanges[i].GetWebsocket()
-				if err != nil {
-					log.Errorf(
-						log.WebsocketMgr,
-						"Exchange %s GetWebsocket error: %s",
-						exchanges[i].GetName(),
-						err,
-					)
-					return
-				}
+		if m.verbose {
+			log.Debugf(log.WebsocketMgr, "Exchange %s websocket support: Yes Enabled: %v",
+				exch.GetName(),
+				common.IsEnabled(exch.IsWebsocketEnabled()))
+		}
 
-				if ws.IsEnabled() {
-					err = ws.Connect()
-					if err != nil {
-						log.Errorf(log.WebsocketMgr, "%v", err)
-					}
+		ws, err := exch.GetWebsocket()
+		if err != nil {
+			log.Errorf(log.WebsocketMgr, "Exchange %s GetWebsocket error: %s",
+				exch.GetName(),
+				err)
+			continue
+		}
 
-					err = m.websocketDataReceiver(ws)
-					if err != nil {
-						log.Errorf(log.WebsocketMgr, "%v", err)
-					}
+		if !ws.IsEnabled() {
+			continue
+		}
 
-					err = ws.FlushChannels()
```

Comment:
Looks like this was removed since last review, could you explain why it was removed?

Reply:
Looks like we didn't need it, it only gets called once, we connect and subscribe above? 🤷 And I totally forgot I did that and should have commented so nice spotting 🦅